### ISSUE TITLE
node-openid is not compatible with squid proxy http -> https

### ISSUE THREAD

**Issue by mitermayer** [havard/node-openid#134]
When behind a proxy server node-openid does not manage to successfully create a request. Causing proxies to return 'ERR_UNSUP_REQ'.

**Comment by mitermayer** [comment]
@havard  If the requests were made with 'request' library that would solve this issue. Is there any plans on tackling that any time soon?

**Comment by mitermayer** [comment]
Other options is applying the fixes from here https://github.com/joyent/node/issues/2474#issuecomment-3484290 

example would be: 

```
var http = require('http');
var https = require('https');

var connectReq = http.request({ // establishing a tunnel
  host: 'localhost',
  port: 3128,
  method: 'CONNECT',
  path: 'github.com:443',
}).on('connect', function(res, socket, head) {
  // should check res.statusCode here
  var req = https.get({
    host: 'github.com',
    socket: socket, // using a tunnel
    agent: false    // cannot use a default agent
  }, function(res) {
    res.setEncoding('utf8');
    res.on('data', console.log);
  });
}).end();
```

**Comment by mitermayer** [comment]
I can create a PR to add support to https proxy

**Comment by havard** [comment]
Thanks for researching, that would be great. And of course, even greater if
you'd be willing to take on the aforementioned request rewrite. :)

-- 
Med vennlig hilsen / Best regards

Håvard Stranden

**Comment by mitermayer** [comment]
I won't have enough time to rewrite the request proxy at the moment, but will add the support for https proxy.

**Comment by havard** [comment]
That's great!

-- 
Med vennlig hilsen / Best regards

Håvard Stranden

**Comment by havard** [comment]
Merged the request rewrite by @sreuter into master now and bumped to/published v1.0.0. Let me know if things don't work for you.
